crypto/src/vrf/schnorrkel/mod.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): S. Afach
+
+use crate::random::{CryptoRng, Rng};
+use merlin::Transcript;
+use schnorrkel::Keypair;
+use serialization::{Decode, Encode};
+
+use self::data::SchnorrkelVRFReturn;
+
+use super::{primitives::VRFReturn, VRFError};
+
+const PUBKEY_LEN: usize = 32;
+const PRIVKEY_LEN: usize = 64;
+
+pub mod data;
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[must_use]
+pub struct SchnorrkelPublicKey {
+    key: schnorrkel::PublicKey,
+}
+
+impl SchnorrkelPublicKey {
+    pub fn from_private_key(private_key: &SchnorrkelPrivateKey) -> Self {
+        SchnorrkelPublicKey {
+            key: private_key.key.to_public(),
+        }
+    }
+
+    pub fn verify_generic_vrf(
+        &self,
+        message: Transcript,
+        vrf_data: &VRFReturn,
+    ) -> Result<(), VRFError> {
+        match vrf_data {
+            VRFReturn::Schnorrkel(d) => self.verify_vrf(message, d),
+        }
+    }
+
+    pub fn verify_vrf(
```

Comment:
This name feels as generic as the "generic" one above in all honesty

Reply:
I have very little confidence about the naming scheme... if you can think of a different naming scheme (and also the design), please go ahead.